### DISCUSSION TITLE
Bonus chance

### DISCUSSION THREAD

**Discussion by typhoon11** [Androz2091/discord-giveaways#384] [Ideas]
**Is your feature request related to a problem? Please describe.**  

No

**Describe the solution you'd like**

It would be really good if you add an option for a bonus chance so that we can add a bonus chance for the users who vote for the bot on any bot listing site or used a referral code.
We can set the percentage of bonus chance like we use the edit, reroll option.

**Comment by reinacchi** [comment]
Did you mean [Bonus Entries](https://github.com/Androz2091/discord-giveaways#bonus-entries)?

**Comment by Nico105** [comment]
/ he  means #258 i guess since he's talking about percentage
only that working with percentage is a little tricky/the accuracy is ~ (it also varies depending on how many other ppl have pecentages).

I would for now recommend the normal bonus entries.
Until we decide what to do with #258 / its algorithm.

**Comment by typhoon11** [comment]
You can leave the bonus chance empty, but if the dev enters a number you can 
``
you could give each user an amount, such as 1, then give other users larger amounts such as 1.25. Users will be ordered so user 1 could have the range 0-1, user 2 could have 1-2.25 and user 3 could have 2.25 - 3.25. The code will pick a random number between 0 and the total and will pick the user depending on which range the number was in. Remove the range and repeat for multiple winners (or just keep the ranges and repeat if duplicates are selected)
``
This solution was given by my friend, if you can understand you can implement it.

**Comment by Nico105** [comment]
The thing here is that we cannot say "x percent greater chance than everyone else" cause if 1 other guy has also even 1 % more then it will be an infinite increasing of ranges.

The only thing we can do is say "x percent greater chance than everyone else while assuming everyone else has no greater chance", means if other ppl have also increased chances then those will not be accurate.


formulas: 
`chanceOf0%(range=1)GreaterChanceUser = 1 * 100 / length`
ex: 10%
_question here: should the person with >% chance get excluded of the length = 10-1=9?_

`rangeForUserWith5%GreaterChance = length * (10 + 5) / 100`
ex range: 1.5
Means for "~5%" greater chance we would need a range of 1.5 for that user
(length=participants was assumed 10)

correct me if i'm false/there is a better way/a way to go around the problem mentioned first.


but i have to say idk if this approach is really better than #258 's algorithm which afaik also takes the other greater chances into account, at least a little bit (i think=im no expert)

_honestly thinking about such things melts my brain_

**Comment by typhoon11** [comment]
Lol true it also melts my brain, I am just wondering now the bot that host giveaways and give the option for 20% more chance are they legit options 🤔
